SwiftRuntimeLibrary/SwiftEquatableProxy.cs

```diff
@@ -32,15 +32,15 @@ static SwiftEquatableProxy ()
 		}
 
 		[UnmanagedCallersOnly]
-		static byte EqFunc (IntPtr oneptr, IntPtr twoptr)
+		static int EqFunc (IntPtr oneptr, IntPtr twoptr)
```

Comment:
As @dalexsoto says, you need to match the size of the native type, otherwise you can end up reading random bits and you'll see "true" when the native code meant "false" (and yes, this happens, I've run into it in the past).

Reply:
The code that uses this always masks off the 0. How would that be an issue?
I mean, I could make everything return `nint` or `nuint` but I figured this is likely to aot into nicer code.

Reply:
Ah, I see, returning an `int` from a function when the caller expects a `bool` should be fine (the opposite - expecting an `int` in the caller when the callee returns a `byte` or `bool` - is not, but that's not the case here).